src/ts/utils/test/range.spec.ts

```diff
@@ -338,4 +348,90 @@ describe("Range utils", () => {
       ]);
     });
   });
+  describe("mapAddedRange", () => {
+    it("should account for a range added before the given range", () => {
+      const incomingRange = { from: 10, to: 15 };
+      const addedRange = { from: 0, to: 5 };
```

Comment:
`nit` I think the names incomingRange and addedRange don't mean much to me and make this harder to parse.

Reply:
I've changed `incomingRange` to `currentRange` – does that make more sense? Might bear some discussion.